com.unity.render-pipelines.universal/Runtime/2D/Clipper.cs

```diff
@@ -0,0 +1,5025 @@
+
+        internal void DoSquare(int j, int k)
+        {
+            double dx = Math.Tan(Math.Atan2(m_sinA,
+                        m_normals[k].X * m_normals[j].X + m_normals[k].Y * m_normals[j].Y) / 4);
+            var pt = new IntPoint(
+                Round(m_srcPoly[j].X + m_delta * (m_normals[k].X - m_normals[k].Y * dx)),
+                Round(m_srcPoly[j].Y + m_delta * (m_normals[k].Y + m_normals[k].X * dx)));
+            pt.NX = m_normals[k].X - m_normals[k].Y * dx; pt.NY = m_normals[k].Y + m_normals[k].X * dx;
+            m_destPoly.Add(pt);
+            pt = new IntPoint(
+                Round(m_srcPoly[j].X + m_delta * (m_normals[j].X + m_normals[j].Y * dx)),
+                Round(m_srcPoly[j].Y + m_delta * (m_normals[j].Y - m_normals[j].X * dx)));
+            pt.NX = m_normals[k].X + m_normals[k].Y * dx; pt.NY = m_normals[k].Y - m_normals[k].X * dx;
+            m_destPoly.Add(pt);
+        }
+
+        //------------------------------------------------------------------------------
+
+        internal void DoMiter(int j, int k, double r)
+        {
+            double q = m_delta / r;
+            var pt = new IntPoint(Round(m_srcPoly[j].X + (m_normals[k].X + m_normals[j].X) * q),
+                Round(m_srcPoly[j].Y + (m_normals[k].Y + m_normals[j].Y) * q));
+            pt.NX = (m_normals[k].X + m_normals[j].X) * q; pt.NY = (m_normals[k].Y + m_normals[j].Y) * q;
+            m_destPoly.Add(pt);
+        }
+
+        //------------------------------------------------------------------------------
+
+        internal void DoRound(int j, int k)
+        {
+            double a = Math.Atan2(m_sinA,
+                    m_normals[k].X * m_normals[j].X + m_normals[k].Y * m_normals[j].Y);
+            int steps = Math.Max((int)Round(m_StepsPerRad * Math.Abs(a)), 1);
+
+            double X = m_normals[k].X, Y = m_normals[k].Y, X2;
+            for (int i = 0; i < steps; ++i)
+            {
+                var pt = new IntPoint(
+                    Round(m_srcPoly[j].X + X * m_delta),
+                    Round(m_srcPoly[j].Y + Y * m_delta));
+                pt.NX = X; pt.NY = Y; pt.N = j; pt.D = 1;
+                m_destPoly.Add(pt);
+                X2 = X;
+                X = X * m_cos - m_sin * Y;
+                Y = X2 * m_sin + Y * m_cos;
+            }
+
+            var pt1 = new IntPoint(
+                Round(m_srcPoly[j].X + m_normals[j].X * m_delta),
+                Round(m_srcPoly[j].Y + m_normals[j].Y * m_delta));
+            pt1.NX = m_normals[j].X; pt1.NY = m_normals[j].Y; pt1.N = j; pt1.D = 1;
+            m_destPoly.Add(pt1);
+        }
+
+        //------------------------------------------------------------------------------
+    }
+
+    class ClipperException : Exception
+    {
+        public ClipperException(string description) : base(description) {}
+    }
+    //------------------------------------------------------------------------------
+} //end ClipperLib namespace
```

Comment:
A few questions regarding this 3rd party source:
* Do we need the review of legal?
* It introduces public APIs under the namespace UnityEngine.Experimental.Rendering.Universal, which I don't think we should do.

Reply:
Good catch. Will remove all public APIs to internal. Not sure regarding license. I think we already use it in Unity/Trunk.

Reply:
Fixed in https://github.com/Unity-Technologies/Graphics/pull/1707/commits/57dfa2a4af15ad19146d1cfb14fc4454719af1ca